### DISCUSSION TITLE
Labels for edges

### DISCUSSION THREAD

**Discussion by bojika** [antvis/G6#5923] [Ideas]
The current release can support only one label per edge.
It would be great to have three labels, at the beginning, in the middle and at the end.
It's useful feature for network topologies, for example (port1, network, port2) or (ip_1, network, ip_2) or (port1, cost, port2).

**Comment by RackweLLizm** [comment]
In my tests on a graph consisting of 4000 nodes/edges, if I turn off the Edge label, there is no freezing on the canvas. When edge labels are active, the canvas becomes incredibly slow. If there are 3, I think it will start to contract at 500, of course I haven't tried it :)

**Comment by Aarebecca** [comment]
The appeal you mentioned does make sense, and we will consider adding this feature in the future, but for now you may only be able to achieve it through custom edges
